src/main/webapp/app/entities/search/search.util.ts

```diff
@@ -0,0 +1,317 @@
+  measuresList.forEach(element => {
+    const measure = features.find(item => {
+      const featureName = element.substring(element.search(/[(]/g) + 1, element.search(/[)]/g));
+      return item.name === trim(featureName);
+    });
+    if (measure) {
+      measures.push(measure);
+    }
+  });
+  return measures;
+};
+
+const setDimensions = (dimensions: readonly IFeature[], visual: IVisualMetadataSet, visualWrap: any) => {
+  dimensions.forEach((element, index) => {
+    if (index === 0) {
+      visual.fields[index].feature = dimensions[index];
+    } else {
+      const fieldType = visualWrap.nextFieldDimension(visual.fields, visual.metadataVisual);
+      const field = {
+        fieldType,
+        feature: dimensions[index],
+        constraint: fieldType.constraint,
+        properties: fieldType.propertyTypes.map(item => {
+          return {
+            propertyType: item.propertyType,
+            value: item.propertyType.defaultValue,
+            type: item.propertyType.type,
+            order: item.order,
+          };
+        }),
+        order: fieldType.order,
+      };
+      visual.fields.push(field);
+    }
+  });
+  return visual;
+};
+
+const setMeasures = (measures: readonly IFeature[], visual: IVisualMetadataSet, visualWrap: any) => {
+  measures.forEach((element, index) => {
+    const fieldType = visualWrap.nextFieldMeasure(visual.fields, visual.metadataVisual);
+    const field = {
+      fieldType,
+      feature: measures[index],
+      constraint: fieldType.constraint,
+      properties: fieldType.propertyTypes.map(item => {
+        return {
+          propertyType: item.propertyType,
+          value: item.propertyType.defaultValue,
+          type: item.propertyType.type,
+          order: item.order,
+        };
+      }),
+      order: fieldType.order,
+    };
+    visual.fields.push(field);
+  });
+  return visual;
+};
+
+export const createVisualFields = (
+  features: readonly IFeature[],
+  dimensionsList: Array<string>,
+  measuresList: Array<string>,
+  view: IViews,
+  visualization: IVisualizations
+) => {
+  let visual = createVisualMetadata(visualization, view, 0);
+  const dimensions = createDimensions(features, dimensionsList);
+  const measures = createMeasures(features, measuresList);
+  const visualWrap = VisualWrap(visual);
+  visual = setDimensions(dimensions, visual, visualWrap);
+  visual = setMeasures(measures, visual, visualWrap);
+  return visual;
+};
+
+const isDynamicDateRange = value => {
+  let conditionType = 'In';
+  for (const element of DYNAMIC_DATE_RANGE_CONFIG) {
+    if (value.toLowerCase() === element.title.toLowerCase()) {
+      conditionType = BETWEEN;
+      break;
+    }
+  }
+  return conditionType;
+};
+
+const isCompareTypes = item => {
+  const compareType: CompareType = {
+    isCompare: false,
+    expression: {
+      feature: '',
+      condition: '',
+      statement: '',
+      featureDataType: '',
+      statements: [],
+    },
+  };
```

Comment:
create compare type model

Reply:
CompareType is already modal 
[https://github.com/DX26-io/data-studio-react/blob/feature/search-query/src/main/webapp/app/entities/search/search.model.ts#L42](url)